packages/ui/src/SelectField.tsx

```diff
@@ -0,0 +1,145 @@
+
+const DropdownIndicator = () => <Icon icon={ChevronDown} ariaHidden size="normal" className={styles.chevron} />
+
+// innerProps set event handling
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ClearIndicator = ({ innerProps }: IndicatorProps<SelectFieldOption<any>>) => {
+  // Visually impaired people will use the keyboard (backspace) to remove the value. We do not want to confuse them by allowing to focus this button.
+  return <IconButton {...innerProps} icon={X} ariaHidden kind="primary" size="normal" className={styles.clear} tabIndex={-1} />
+}
+
+export type SelectFieldOption<V = string> = {
+  label: string
+  value: V
+}
+
+export type SelectFieldCoreStaticProps = {
+  name: string
+  onBlur?: (e: FocusEvent<HTMLElement>) => void
+  error?: string
+}
+export type SelectFieldCoreDynamicProps<V> =
+  | {
+      isClearable: true
+      value: SelectFieldOption<V> | null
+      onChange: (newValue: SelectFieldOption<V> | null) => void
+    }
+  | {
+      isClearable?: false
+      value: SelectFieldOption<V>
+      onChange: (newValue: SelectFieldOption<V>) => void
+    }
+export type SelectFieldExtraProps<V> = {
+  options: SelectFieldOption<V>[]
+  label: string
+  required?: boolean
+  helperText?: string | ReactElement
+  className?: string
+  errorClassName?: string
+  placeholder?: string
+  menuPortalTarget?: 'body' | 'self' | HTMLElement | null
+} & DataTestProp &
+  Pick<InputHTMLAttributes<HTMLElement>, 'autoFocus' | 'disabled' | 'autoComplete'> &
+  Pick<ReactSelectProps, 'isSearchable' | 'menuIsOpen' | 'menuPlacement'>
+
+export type SelectProps<V> = SelectFieldCoreStaticProps & SelectFieldCoreDynamicProps<V> & SelectFieldExtraProps<V>
+
+const getMenuContainer = (menuPortalTarget: 'body' | 'self' | HTMLElement | null): HTMLElement | null | undefined => {
+  if (menuPortalTarget == 'body') {
+    return document.body
+  }
+  if (menuPortalTarget == 'self') {
+    return undefined
+  }
+  return menuPortalTarget
+}
+
+export const SelectField = <V,>({
+  'data-test': dataTest,
+  className,
+  error,
+  errorClassName,
+  isClearable,
+  disabled,
+  isSearchable = false,
+  label,
+  name,
+  required,
+  value,
+  onChange,
+  helperText,
+  menuPortalTarget = 'body',
+  ...rest
+}: SelectProps<V>): ReactElement<SelectProps<V>> => {
+  const idRef = useRef(uuid())
+
+  useIsomorphicLayoutEffect(() => {
+    const menuContainer = getMenuContainer(menuPortalTarget)
+
+    if (menuContainer && !menuContainer.className.includes(styles.menuContainer)) {
+      menuContainer.className = `${menuContainer.className} ${styles.menuContainer}`
+    }
+  }, [menuPortalTarget])
+
+  return (
+    <div
+      data-test={dataTest}
+      className={cn(
+        styles.container,
+        {
+          [styles.disabled]: disabled,
+          [styles.hasError]: error,
+          [styles.empty]: !value,
+        },
+        // Menu container is either this select itself or document.body
```

Comment:
> Menu container is either this select itself or document.body
According to `getMenuContainer` it could be any element. Or am I mixing two different things here?